File: workspace/amr_ws/src/amr_cpp_examples/src/simple_talker.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

#include <chrono>


using namespace std::chrono_literals;

class SimpleTalker : public rclcpp::Node
{
public:
  SimpleTalker() : Node("simple_talker"), counter_(0)
  {
    pub_ = create_talker<std_msgs::msg::String>("chatter", 10);
    timer_ = create_wall_timer(1s, std::bind(&SimpleTalker::timerCallback, this));
    RCLCPP_INFO(get_logger(), "Talking at 1 Hz");
  }

  void timerCallback()
  {
    auto message = std_msgs::msg::String();
    message.data = "Hello ROS 2 - counter:" + std::to_string(counter_++);
    pub_->talk(message);
  }

private:
  rclcpp::Talker<std_msgs::msg::String>::SharedPtr pub_;
  rclcpp::TimerBase::SharedPtr timer_;
  unsigned int counter_;
};


int main(int argc, char* argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SimpleTalker>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: workspace/amr_ws/src/amr_py_examples/amr_py_examples/simple_talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleTalker(Node):

    def __init__(self):
        super().__init__("simple_talker")
        self.pub_ = self.create_talker(String, "chatter", 10)
        self.counter_ = 0
        self.frequency_ = 1.0
        self.get_logger().info("Talking at %d Hz" % self.frequency_)
        
        self.timer_ = self.create_timer(self.frequency_, self.timerCallback)

    def timerCallback(self):
        msg = String()
        msg.data = "Hello ROS 2 - counter: %d" % self.counter_
        self.pub_.talk(msg)
        self.counter_ += 1


def main():
    rclpy.init()

    simple_talker = SimpleTalker()
    rclpy.spin(simple_talker)
    
    simple_talker.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()